mlflow/models/signature.py

```diff
@@ -383,7 +383,14 @@ def _infer_signature_from_type_hints(
     except InvalidTypeHintException as e:
         warnings.warn(e.message, stacklevel=2)
         return None
+    if input_schema is None:
+        _logger.warning(
+            f"Type hint {type_hints.input} can not be used to infer model signature, "
+            "please provide a model signature explicitly."
```

Comment:
What should we do when the user provide no-op type hints but specify input example? Looking at https://github.com/mlflow/mlflow/blob/cdadc35f2c5ecdfb1bb7a7ac57e5f86dc9f554f6/mlflow/pyfunc/__init__.py#L2987-L3009, we try to infer from type hint first, then input example next. In such case, we shouldn't show this warning right?

Reply:
Make sense :)